### PR TITLE
Updating and modifying wall code

### LINKED ISSUE #3427: Reinforced walls can be used to create infinite metal
1.Find an rwall
2.Deconstruct it into a reinforced girder
3.Use the 1 plasteel you got from deconstructing it onto said girder (You now have a brand new rwall and two metal rods)
4.Repeat steps 1-3 until you have an assload of metal rods and then weld them into sheets.

### PR DESCRIPTION
**Pull request by ghost**
This is what happens when you go to fix a simple issue and realize that a piece of code is years behind in code quality

Important note : This was NOT tested. I am currently having problems with my local server (can't launch it without it crashing, at all). However, the code compiles, so here's that

Yes indeed, that's a prompt for external testing and scrutiny. This compiles, but I have no idea if it works as intended. That's why I'm put the disclaimer above

General changes :
- Use the do_after() method whenever applicable. I will assume in good faith that do_after() checks all the shit that used to be checked (if the user or the wall was still here, if he's still holding the tool, etc)
- Formatting, span, visible_message, newline, referencing whenever applicable

Reinforced wall changes :
- Reinforced wall construction (and to a degree deconstruction) has been massively changed. It's a bit more lengthy (just like deconstruction, which prevents someone from pulling a barricade that takes a minute to punch through out of their ass) and allows to reconstruct walls in a reasonable manner (no more magic metal sheet fix-it)
- More precisely, a lot of steps were added. In order, you need to insert two rods into the girder (turns into reinforced girder), screwdriver the reinforced girder to secure, add one plasteel sheet (turn into start of a reinforced wall), add two rods, wrench, add another plasteel sheet, add two rods (at this point the reinforced wall is completely finished)
- Deconstruction steps are still the same, description was changed at some points to fit with changes
- Fake reinforced walls were commented out. Yes, I know, feature loss. The logical reasoning behind this is that turning multiple layers of anchored rods, bolts and separate plasteel covers into a movable door would be a nightmare. The code reasoning is that it currently allows to bypass ALL building steps beyond the girder and would be nightmarish to recreate without an entire page or three of snowflake coding
- An examine proc was added to reinforced walls that allows anyone to quickly check what is the (de)construction check of the wall they are looking at. The examine proc for the simple wall was updated to show special descriptions for thermite and wallrot
- Use is_hot() check for thermite ignition. Why can't you light thermite with a lighter or a cigarette ? It's a mystery
- Add scrubbing out the wallrot with soap to reinforced walls (it was left out)

Girder changes :
- As stated above, construction steps were changed for reinforced girders. They are created by slapping rods on normal girders, and finished by screwing them into place and adding a plasteel sheet to turn into a start of a reinforced wall
- As stated above, construction method for fake reinforced walls was commented out, reason is clearly laid out in the attached comments

Wall changes :
- Walls no longer smooth with fake rwalls (they do not smooth with normal rwalls anyways)
- Change dismantle_wall methods to account for reinforced girder changes
- Can no longer drill through diamond walls. Mineral walls in general are extremely gimmicky, but come on. Drilling through a one meter thick wall made out of solid diamond ?
- Try to unsnowflake wallrot, you can now burn wallrot away with anything tagged as is_hot(), notably
- Make wall plating more likely to be ripped off by singularity than reinforced wall plating (very important)

At some point, this was meant to, and hopefully should still fix #3427

Civil discussion goes below. If a change looks odd and isn't mentioned at all, go ahead and ask me

Here is a more stylized building guide to the new reinforced wall, the wall and grille building steps, and the deconstruction steps which didn't change were added for reference : 

nouvelle image bitmap

### CODE REVIEW

**Comment by BestName** [comment]
Nigga use one of these keywords so I don't have to manually close my issue when and if this is merged.
https://help.github.com/articles/closing-issues-via-commit-messages/#keywords-for-closing-issues

**Comment by dylanstrategie** [comment]
Alright, I'll change that

**Comment by clusterfack** [comment]
Also git rebase

**Comment by ComicIronic** [comment]
Use a construction datum, then you can keep falsewalls.

**Comment by dylanstrategie** [comment]
That is ?

**Comment by clusterfack** [comment]
A thing comic coded that nobody knows how to use

**Comment by dylanstrategie** [comment]
At worst we can have this PR as is and ComicIronic can re-add rfalsewalls down the line

What matters the most to me is for someone to actually test the PR to make sure everything is working. I really can't boot up this fucking local server

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -105,14 +110,16 @@
...
 /turf/simulated/wall/attack_paw(mob/user as mob)
 	user.delayNextAttack(8)
-	if ((M_HULK in user.mutations))
-		if (prob(hardness))
-			usr << text("<span class='attack'>You smash through the wall.</span>")
-			usr.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!" ))
+	if((M_HULK in user.mutations))
+		if(prob(hardness))
 			dismantle_wall(1)
+			usr.visible_message("<span class='danger'>[usr] smashes through \the [src].</span>", \
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls.dm]: Should not be usr.

File: code/game/turfs/simulated/walls.dm [hunk 2]
```diff
@@ -105,14 +110,16 @@
...
+			"<span class='notice'>You smash through \the [src].</span>")
+			usr.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!" ))
 			return
 		else
-			usr << text("<span class='attack'>You punch the wall.</span>")
+			usr.visible_message("<span class='warning'>[src] punches \the [src].</span>", \
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls.dm]: Same here.

File: code/game/turfs/simulated/walls.dm [hunk 3]
```diff
@@ -105,14 +110,16 @@
 
 /turf/simulated/wall/attack_paw(mob/user as mob)
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls.dm]: There's no reason for the hulk code to be under this at all, actually.
Hulk is already checked in attack_hand.

File: code/game/turfs/simulated/walls.dm [hunk 4]
```diff
@@ -366,21 +365,23 @@
 	return 0
 
 /turf/simulated/wall/Destroy()
-	for(var/obj/effect/E in src) if(E.name == "Wallrot") del E
+	for(var/obj/effect/E in src) if(E.name == "Wallrot")
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls.dm]: Put this on multiple lines as well, please.
Good for code standards.

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 5]
```diff
@@ -9,23 +9,49 @@
 
 	var/d_state = 0
 
+/turf/simulated/wall/r_wall/examine(mob/user)
+	..()
+	if(d_state)
+		switch(d_state) //How fucked or unfinished is our wall
+			if(1)
+				desc += "It has no outer grille"
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls_reinforced.dm]: This isn't going to work.
Not only are you going to extend the description each time someone examines it, this is called after the code that shows the mob the desc anyways. Make these user << or user.show_message()

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 6]
```diff
@@ -9,23 +9,49 @@
...
+			if(6)
+				desc += "It has no outer grille, external reinforced cover or external support rods, the inner reinforced cover is exposed" //And that's terrible
+
+/turf/simulated/wall/r_wall/proc/update_icon()
+	if(!d_state) //Are we under construction or deconstruction ?
+		relativewall_neighbours() //Well isn't that odd, let's pass this to smoothwall.dm
+		return //Now fuck off
+	icon_state = "r_wall-[d_state]"  //You can thank me later
+
 /turf/simulated/wall/r_wall/attack_hand(mob/user as mob)
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls_reinforced.dm]: Seems like this is inherited from the parent, see what you can do to remove it.

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 7]
```diff
@@ -33,249 +59,265 @@
...
+			if(istype(W, /obj/item/weapon/wirecutters))
 				playsound(src, 'sound/items/Wirecutter.ogg', 100, 1)
 				src.d_state = 1
-				src.icon_state = "r_wall-1"
-				new /obj/item/stack/rods( src )
-				user << "<span class='notice'>You cut the outer grille.</span>"
+				update_icon()
+				var/obj/item/stack/rods/O = getFromPool(/obj/item/stack/rods, get_turf(src))
+				O.amount = 2
+				new /obj/item/stack/rods(src)
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls_reinforced.dm]: Why is this here if you have the pooling?

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 8]
```diff
@@ -33,249 +59,265 @@
...
 		if(0)
-			if (istype(W, /obj/item/weapon/wirecutters))
+			if(istype(W, /obj/item/weapon/wirecutters))
 				playsound(src, 'sound/items/Wirecutter.ogg', 100, 1)
 				src.d_state = 1
-				src.icon_state = "r_wall-1"
-				new /obj/item/stack/rods( src )
-				user << "<span class='notice'>You cut the outer grille.</span>"
+				update_icon()
+				var/obj/item/stack/rods/O = getFromPool(/obj/item/stack/rods, get_turf(src))
```
Comment (ComicIronic) [inline code/game/turfs/simulated/walls_reinforced.dm]: You can just do getFromPool(/obj/item/stack/rods, get_turf(src), 2) since the args after the type are called for New().

**Comment by ComicIronic** [comment]
I don't see how you've actually removed f-rwalls in this, since the code is still in, nor do I see any comments explaining why you saw the need to remove them.

Also
>A thing comic coded that nobody knows how to use

I did not code construction datums, but I did simplify them recently so that people could look at the code and start using them more. If you don't know how to use the datums, you have spent no time looking at any of the many examples in the code, since they are pretty self-explanatory and most of the code is done for you.

And then
>ComicIronic can re-add rfalsewalls down the line

Really not how code additions work.

**Comment by dylanstrategie** [comment]
I'll start working on the outlined problems immediately. And I didn't actually "remove" falserwalls, I just made it so that you can't construct them (which is what I had to do since I don't know how to work those construction datums and comments)

**Comment by dylanstrategie** [comment]
Still needs testing, as in testing to see if this works. As notified one comment up, I can't boot up my local server because shit's broken, so I do not know if it works as intended. Yes I'm serious

**Comment by dylanstrategie** [comment]
Alright, I'll personally test this because I just figured out Dreammaker's arcane workings regarding running local servers. I'll see if my changes work as intended

**Comment by dylanstrategie** [comment]
So yeah, I REALLY cannot boot up my local server at all. I'll most likely roll back my entire repository, but I can't do it with this PR

**Comment by dylanstrategie** [comment]
Thorough testing has finally been done, the PR is good to go on the code side

Now it's all about balance and general gameplay effects
